x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Verifier.java

```diff
@@ -208,6 +212,48 @@ else if (p instanceof Lookup lookup) {
         return failures;
     }
 
+    private static void checkMetadataScoreNameReserved(LogicalPlan p, Set<Failure> failures) {
+        // _score can only be set as metadata attribute
+        if (p.inputSet().stream().anyMatch(a -> MetadataAttribute.SCORE.equals(a.name()) && (a instanceof MetadataAttribute) == false)) {
+            failures.add(fail(p, "`" + MetadataAttribute.SCORE + "` is a reserved METADATA attribute"));
+        }
+    }
+
+    /**
+     * Check for patterns like this: `... | WHERE match(..) | ... | EVAL whatever = _score | ... | WHERE match() ... `.
+     * This is currently disallowed because of the fact that both such match/qstr/etc. would get pushed down together (as an AND) to ES.
+     * Hence any such supposedly "intermediate" _score (for the first query) would be meaningless, as it would be identical to the
+     * (final) _score of the combined queries.
+     */
+    private static void checkScoreManipulationBetweenMultipleFulltextExpressions(LogicalPlan p, Set<Failure> failures) {
```

Comment:
I see your point, just again _score is a side effect of executing text matching functions, a user expects to be executing things "in pipes", so this check is meant to be temporary until we make it possible to execute queries one after the other and isolate scores when the user asks to do so.

Reply:
> a user expects to be executing things "in pipes"

That's why IMO we should document the way this works, rather then restrict through the verifier. It could be that if the user wants to isolate the effects of a match, we'll need other isolation mechanisms, like subqueries.

If you execute `... | MATCH ... | WHERE _score > ... | MATCH ... | WHERE _score > ...`, that won't work "in pipes" either though I suspect it'd now run fine.

Reply:
I agree on evaluating the use of subqueries for this use case, the check in the Verifier are only meant to be temporary until we provide support for such kind of use cases. I am collecting all of this feedback in a [meta-issue](#116599), if that can help, so that we can eventually progress this incrementally.

Reply:
ok, I'll lift this restriction for this PR, I've added another task for this to be further evaluated in the [meta-issue](https://github.com/elastic/elasticsearch/pull/113120#116599) (_
clarify valid scenarios for _score manipulation_).